File: sdks/dotnet/src/Croniq.Runner.Sdk/Protocol/TriggerResponse.cs
using System.Text.Json.Serialization;

namespace Croniq.Runner.Sdk.Protocol;

/// <summary>
/// Wire response of <c>POST /v1/trigger</c>. <c>deduplicated</c> is sent by
/// servers that support trigger idempotency keys and <c>coalesced</c> by
/// servers that support the <c>coalesce</c> job directive; older servers omit
/// either and the field defaults to <c>false</c>.
/// </summary>
internal sealed record TriggerResponse(
    [property: JsonPropertyName("execution_id")] string ExecutionId,
    [property: JsonPropertyName("queued")] int Queued,
    [property: JsonPropertyName("deduplicated")] bool Deduplicated = false,
    [property: JsonPropertyName("coalesced")] bool Coalesced = false);

File: sdks/java/core/src/main/java/io/croniq/runner/protocol/TriggerResponse.java
package io.croniq.runner.protocol;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Wire response of {@code POST /v1/trigger}.
 *
 * <p>{@code deduplicated} is sent by servers that support trigger idempotency
 * keys and {@code coalesced} by servers that support the {@code coalesce} job
 * directive; older servers omit either, and Jackson leaves the primitive at its
 * default ({@code false}) — the forward-compatible parse contract the
 * conformance suite pins.
 */
public record TriggerResponse(
        @JsonProperty("execution_id") String executionId,
        @JsonProperty("queued") int queued,
        @JsonProperty("deduplicated") boolean deduplicated,
        @JsonProperty("coalesced") boolean coalesced) {}
